cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -530,6 +539,14 @@ protected PrincipalAuthenticationResult notifyRemotePrincipalAuthenticated(
             logger.error("[CAS XSLT] Missing names: username={}, institution={}", username, institutionId);
             throw new InstitutionLoginFailedAttributesMissingException("Missing user's names");
         }
+        String department = "";
+        if (departmentRaw.isEmpty()) {
+            logger.warn("[CAS XSLT] Missing department: fullname={}, username={}, institution={}", fullname, username, institutionId);
+        } else {
+            department = this.retrieveDepartment(departmentRaw, eduPerson);
+        }
+        normalizedPayload.put("department", department);
```

Comment:
`department` should be under `user` which is under `provider`. See this [merge commit](https://github.com/CenterForOpenScience/osf.io/commit/9b883a570ad588aecec4bf064c21bf15c6f8088f#diff-dfebf3b9b92620b1c35c25de8d8f7a46R87) on the OSF side.